internal/elasticsearch/enrich/policy.go

```diff
@@ -0,0 +1,184 @@
+package enrich
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/elasticsearch"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceEnrichPolicy() *schema.Resource {
+	policySchema := map[string]*schema.Schema{
+		"name": {
+			Description: "Name of the enrich policy to manage.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+		},
+		"policy_type": {
+			Description:  "The type of enrich policy, can be one of geo_match, match, range.",
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringInSlice([]string{"geo_match", "match", "range"}, false),
+		},
+		"indices": {
+			Description: "Array of one or more source indices used to create the enrich index.",
+			Type:        schema.TypeSet,
+			Required:    true,
+			ForceNew:    true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"match_field": {
+			Description: "Field in source indices used to match incoming documents.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 255),
+			),
+		},
+		"enrich_fields": {
+			Description: "Fields to add to matching incoming documents. These fields must be present in the source indices.",
+			Type:        schema.TypeSet,
+			Required:    true,
+			ForceNew:    true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"query": {
+			Description:  "Query used to filter documents in the enrich index. The policy only uses documents matching this query to enrich incoming documents. Defaults to a match_all query.",
+			Type:         schema.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsJSON,
```

Comment:
```suggestion
			ValidateFunc: validation.StringIsJSON,
			DiffSuppressFunc: utils.DiffJsonSuppress,
```

Should we be suppressing a diff of equivalent JSON objects here?